File: Frontend/src/app/interfaces/ship-quick-reservation-response.ts
import { ResponseUtility } from "./response-utility";

export interface ShipQuickReservationResponse {
    id: string;
    startTime: Date;
    endTime: Date;
    validUntil: Date;
    price: string;
    shipId: string
    guestLimit: string;
    utilities: ResponseUtility[];
    reserved: boolean;
}

File: demo/src/main/java/com/example/demo/dto/ShipQuickReservationResponse.java
package com.example.demo.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Set;

@AllArgsConstructor
@NoArgsConstructor
@Setter
@Getter
public class ShipQuickReservationResponse {
    private String id;
    private String startTime;
    private String endTime;
    private String validUntil;
    private String price;
    private String shipId;
    private String guestLimit;
    private Set<ResponseUtility> utilities;
    private boolean isReserved;
}
